### ISSUE TITLE
🦠 Model Request: Finetuned ImageMol Model for GPCR assays from ChEMBL datasets

### ISSUE THREAD

**Issue by DhanshreeA** [ersilia-os/ersilia#571]
### Model Name

imagemol-gpcr

### Model Description

  Representation Learning Framework that utilizes molecule images for encoding molecular inputs as machine readable vectors for downstream tasks such as bio-activity prediction, drug metabolism analysis, or drug toxicity prediction. The approach utilizes transfer learning, that is, pre-training the model on massive unlabeled datasets to help it in generalizing feature extraction and then fine tuning on specific tasks. The is a regression model and will be fine tuned on top 10 GPCR datasets with the largest number of reported ligands from Chembel datasets.

### Slug

image-mol-gpcr

### Tag

GPCR,regression

### Publication

https://www.nature.com/articles/s42256-022-00557-6

### Source code

https://github.com/HongxinXiang/ImageMol

### License

MIT License

**Comment by DhanshreeA** [comment]
The dataset consists of 10 assays and a pre-trained ImageMol will be fine tuned per assay and together they will be incorporated into Ersilia as a unified model.
![image](https://user-images.githubusercontent.com/7213136/213667052-84c12ca0-00cc-4d4b-9cde-0e342f8bb095.png)

**Comment by DhanshreeA** [comment]
@GemmaTuron @miquelduranfrigola I understand what GPCRs basically are - they're proteins which exists on cell membranes and kind of exist as an API for cells, ie based on the input, some response is activated within the cell. These inputs are "ligands", basically some small molecule (which comes from a drug). What I don't understand is why is this a regression task and not a classification task, ie which question does this answer:

Does molecule bind to protein? Or;
What degree does this molecule bind to protein? (If it's this, then what the unit of measurement for these values?)

**Comment by DhanshreeA** [comment]
> The dataset consists of 10 assays and a pre-trained ImageMol will be fine tuned per assay and together they will be incorporated into Ersilia as a unified model. ![image](https://user-images.githubusercontent.com/7213136/213667052-84c12ca0-00cc-4d4b-9cde-0e342f8bb095.png)

On a second thought, the authors have setup the code base to have as many neurons in the final fully connected layer as there [are tasks](https://github.com/HongxinXiang/ImageMol/blob/master/model/cnn_model_utils.py#L22). So, given ten assays, the final FC layer will have 10 neurons. Consequently there are [metric utilities](https://github.com/HongxinXiang/ImageMol/blob/master/model/evaluate.py#L88) for multi task setting as well.
I can obviously adapt to training a model per assay but perhaps a single model with N-neuros in the FC layer makes more sense, especially since this increases the data available for the model to be trained on.

**Comment by DhanshreeA** [comment]
Waiting for the authors to get back with fine tuned models for these assays as reproducing the results from the paper has been challenging because the exact choice of hyper parameters is not known, and takes some experimentation when considering different possible values from the range of values the authors have provided in their supplementary materials.

**Comment by DhanshreeA** [comment]
Have pinged the authors again and they have replied on email ( @miquelduranfrigola ) is CC'd. Hopefully they will share the model checkpoints soon.

Same for #572

**Comment by DhanshreeA** [comment]
Hi @GemmaTuron could you please add me to this repository? the authors have finally shared the models and since this repo is old, it doesn't have the mock.csv tracked in LFS. Alternatively, you could just update the mock.csv in this repo.

**Comment by DhanshreeA** [comment]
> @GemmaTuron @miquelduranfrigola I understand what GPCRs basically are - they're proteins which exists on cell membranes and kind of exist as an API for cells, ie based on the input, some response is activated within the cell. These inputs are "ligands", basically some small molecule (which comes from a drug). What I don't understand is why is this a regression task and not a classification task, ie which question does this answer:
> 
> Does molecule bind to protein? Or; What degree does this molecule bind to protein? (If it's this, then what is the unit of measurement for these values?)

@GemmaTuron could you please answer this? It will help me update the Inference section better as well.

**Comment by GemmaTuron** [comment]
Hi @DhanshreeA !

The model predicts Drug - Protein binding activity. Drug-protein binding can be measured in different manners, and I don't see much explanation in the paper indeed. What is the headers of the raw datasets? This might provide some more information.
For the interpretation, I'd say:
`Binding activity of the input molecule to the following GPCRs: ...` changing the ... for the GPCR acronyms

**Comment by DhanshreeA** [comment]
> Hi @DhanshreeA !
> 
> The model predicts Drug - Protein binding activity. Drug-protein binding can be measured in different manners, and I don't see much explanation in the paper indeed. What is the headers of the raw datasets? This might provide some more information. For the interpretation, I'd say: `Binding activity of the input molecule to the following GPCRs: ...` changing the ... for the GPCR acronyms

Thanks for the suggestion Gemma. The headers of the raw datasets (that at least the authors have provided) are the same as [here](https://github.com/ersilia-os/ersilia/issues/571#issuecomment-1398154771) 
From the supplementary materials, they mention:
> The top 10 G protein coupled receptors (GPCRs) datasets with the largest number of reported ligands from ChEMBL database (https://www.ebi.ac.uk/chembl/) are used to predict drug-protein binding affinity (regression task)

I looked up one of the receptors (AA1R - Adenosine Receptor A1) on [ChEMBL](https://www.ebi.ac.uk/chembl/target_report_card/CHEMBL3688/) and it seems like the values maybe some sort of binding efficiency index. I could be wrong though.

**Comment by DhanshreeA** [comment]
However @GemmaTuron, I have another question - what is an appropriate tag for this? :sweat_smile: In my understanding this would come under Target identification, what do you think?

**Comment by GemmaTuron** [comment]
The ChEMBL link is very helpful thanks @DhanshreeA !
It seems the measure they are using is the BEI (BEI = (pKi, pKd, or pIC50) / (molecular weight, kDa)) or SEI (or combination of both?). To make sure we are indicating the right measurement, could you check if the output values for a few molecules make sense? (they are in the same range as the ones indicated in ChEMBL --maybe take molecules with high ligand efficiency and check what they score)

Tags: 
GPCR (this should be added in the "Targets" file, under hub/content/metadata) -- can you make a new PR to include it?
Target identification

**Comment by DhanshreeA** [comment]
Hi @GemmaTuron that's a good idea and I tested the model for AA1R against a few compounds from the ChEMBL page and here are the results: 
- [ CHEMBL4533718](https://www.ebi.ac.uk/chembl/compound_report_card/CHEMBL4533718/): model 7.3, reported SEI 7.51
- [CHEMBL4436786](https://www.ebi.ac.uk/chembl/compound_report_card/CHEMBL4436786/): model 7.62, reported SEI 7.97
- [CHEMBL284969](https://www.ebi.ac.uk/chembl/compound_report_card/CHEMBL284969/): model 6.96, reported SEI 7.71
- [CHEMBL1877326](https://www.ebi.ac.uk/chembl/compound_report_card/CHEMBL1877326/) model 6.81, reported SEI 7.85
- [CHEMBL68738](https://www.ebi.ac.uk/chembl/compound_report_card/CHEMBL68738/) model 7.43, reported SEI 7.69

I think it's also worth noting that the model's RMSE across this assay [has been reported to be](https://static-content.springer.com/esm/art%3A10.1038%2Fs42256-022-00557-6/MediaObjects/42256_2022_557_MOESM9_ESM.pdf) **0.711±0.012**. Across these five compounds that I've tested (while a small set), the RMSE comes up to ~0.62 which seems to be consistent with the results reported, and is also a strong indicator that the measure being used is likely SEI.

**Comment by DhanshreeA** [comment]
Associated PR: https://github.com/ersilia-os/eos93h2/pull/1
This model is ready to be tested by others.

**Comment by GemmaTuron** [comment]
Sounds Good thanks Dhanshree!
Please update the interpretation accordingly. I'll close this issue
